src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -560,7 +577,8 @@ void ObjectSynchronizer::enter_for(Handle obj, BasicLock* lock, JavaThread* lock
 
 void ObjectSynchronizer::enter(Handle obj, BasicLock* lock, JavaThread* current) {
   assert(current == Thread::current(), "must be");
-  CHECK_THROW_NOSYNC_IMSE(obj);
+  JavaThread* THREAD = current;
+  CHECK_THROW_NOSYNC_IE(obj);
```

Comment:
We never get here.  The c1 slow path and interpreter code check for inline types before calling this.

I added an assert.